saleor/graphql/discount/mutations/voucher/voucher_create.py

```diff
@@ -98,10 +122,19 @@ class Meta:
 
     @classmethod
     def clean_input(cls, info: ResolveInfo, instance, data, **kwargs):
-        code = data.get("code", None)
-        if code == "":
+        cls.clean_codes(data)
+        cleaned_input = super().clean_input(info, instance, data, **kwargs)
+
+        return cleaned_input
+
+    @classmethod
+    def _clean_old_code(cls, data):
+        # Deprecated in 3.0, remove in 4.0
```

Comment:
Have you created a gh issue for that? It should be assigned to Saleor4.0 Milestione: https://github.com/saleor/saleor/milestone/80

Reply:
We have planty of places with deprecated data in code. All fields with `DEPRECATED_IN` label so I don't think it's required to create a ticket for each deprecation. I suppose we will drop all deprecated fields at once. 
And this method is for cleaning deprecated field data.